src/package/package.ts

```diff
@@ -194,6 +208,64 @@ export class Package {
     return convertPackage(pkgId, connection, options, project);
   }
 
+  /**
+   * Download the metadata files for a previously published package version, convert them to source format, and put them into a new project folder within the sfdx project.
+   *
+   * @param project
+   * @param options {@link PackageVersionMetadataDownloadOptions}
+   * @param connection
+   * @returns
+   */
+  public static async downloadPackageVersionMetadata(
+    project: SfProject,
+    options: PackageVersionMetadataDownloadOptions,
+    connection: Connection
+  ): Promise<PackageVersionMetadataDownloadResult> {
+    // Validate the destination path is suitable to extract package version metadata (must be new or empty)
+    const destinationFolder = options.destinationFolder || 'force-app/';
+    const destinationPath = path.join(project.getPath(), destinationFolder);
```

Comment:
Possibly confusing as it would make a relative directory inside their project dir. I suppose we could raise an error if the path starts with `path.sep`.

Reply:
Now raising an error if an absolute path is given